tests/ui/errors/remap-path-prefix-sysroot.with-remap.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0507]: cannot move out of `self.thread` which is behind a mutable reference
+  --> remapped/tests/ui/errors/remap-path-prefix-sysroot.rs:14:9
+   |
+LL |         self.thread.join().unwrap();
+   |         ^^^^^^^^^^^ ------ `self.thread` moved due to this method call
+   |         |
+   |         move occurs because `self.thread` has type `JoinHandle<()>`, which does not implement the `Copy` trait
+   |
+note: `JoinHandle::<T>::join` takes ownership of the receiver `self`, which moves `self.thread`
+  --> remapped/library/std/src/thread/mod.rs:LL:COL
```

Comment:
This test fails because the path emitted in `note` isn't being remapped. @b-naber do you know where in the code this path is emitted? It's behind too many layers of diagnostic abstractions I couldn't find it.

Reply:
The culprit is here, where all diagnostic paths are preferring non-remapped version
https://github.com/rust-lang/rust/blob/21cce21d8c012f14cf74d5afddd795d324600dac/compiler/rustc_span/src/source_map.rs#L1088-L1090

Reply:
I'm suspecting `JsonEmitter` from https://github.com/rust-lang/rust/blob/21cce21d8c012f14cf74d5afddd795d324600dac/compiler/rustc_errors/src/json.rs#L95-L99 which is constructed by an `EarlyErrorHandler` is being used to emit this error, even though it's not an early error

Reply:
Ah so this has nothing to do with where the `note` is emitted. It was because translated sysroot paths are not subjected to remapping, since they are paths imported from other crates which are not currently further remapped, regardless of `--remap-path-prefix` (see https://github.com/rust-lang/rust/issues/66251#issuecomment-1851941782). I added special case code for translated sysroot paths so they are further remapped as the RFC specified, and this test passes.